python/tvm/relax/frontend/nn/spec.py

```diff
@@ -322,15 +324,18 @@ def _extern_modules() -> List[Tuple[str, List[str]]]:
         # pylint: enable=protected-access
 
         params = _params()
-        effects = _effects()
+        effects = _effects() if debug else None
```

Comment:
Let's only set `io_effect` to None if `debug` flag is off, because there could be other effects, for example, kv cache

Reply:
by setting `self.io_effect` to None in `SpecBuilder.__init__`

Reply:
This is great feedback and makes perfect sense. I've updated the code so that debug only turns off io effects based on your suggestion. Can you give it another look and see if this seems reasonable to you?